### PR TITLE
Fixing NPE in auto-configuration

### PR DESCRIPTION
**Pull request by yevster**
Fixing code in `SpringContextAutoConfiguration` that can result in Null Pointer Exceptions when the `spring.cloud.azure.resource-group` parameter is set in several Spring Boot starters.

### CODE REVIEW

**Review by mnriem — APPROVED** [review]

**Comment by mnriem** [comment]
LGTM

**Review by andrewchoi5 — APPROVED** [review]

File: sdk/spring/azure-spring-cloud-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfigurationTest.java [hunk 1]
```diff
@@ -3,68 +3,104 @@
...
 
     @Test
     public void testRequiredAzureProperties() {
         this.contextRunner.withPropertyValues("spring.cloud.azure.credentialFilePath=credential")
-            .withPropertyValues("spring.cloud.azure.resourceGroup=group1").run(context -> {
-                assertThat(context).hasSingleBean(AzureProperties.class);
-                assertThat(context.getBean(AzureProperties.class).getCredentialFilePath()).isEqualTo("credential");
-                assertThat(context.getBean(AzureProperties.class).getResourceGroup()).isEqualTo("group1");
-            });
+                .withPropertyValues("spring.cloud.azure.resourceGroup=group1").run(context -> {
```
Comment (chenrujun) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfigurationTest.java:54]: Please do not include unnecessary change.
Maybe this file can help you: https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/spring/AzureSpringCodeStyle.xml

File: sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java [hunk 2]
```diff
@@ -37,8 +40,8 @@
 @ConditionalOnProperty(prefix = "spring.cloud.azure", value = { "resource-group" })
 public class AzureContextAutoConfiguration {
 
-    private static final String PROJECT_VERSION =
-        AzureContextAutoConfiguration.class.getPackage().getImplementationVersion();
+    private static final String PROJECT_VERSION = AzureContextAutoConfiguration.class.getPackage()
+            .getImplementationVersion();
```
Comment (chenrujun) [inline sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:44]: Same in this file.

File: sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java [hunk 3]
```diff
@@ -58,17 +62,23 @@ public ResourceManagerProvider resourceManagerProvider(Azure azure, AzurePropert
...
+    }
 
-        String subscriptionId = Optional.of(azureProperties.getSubscriptionId())
-                                        .orElseGet(credentials::defaultSubscriptionId);
-        return Azure.authenticate(restClient, credentials.domain())
-                    .withSubscription(subscriptionId);
+    @VisibleForTesting
+    protected Azure authenticateToAzure(RestClient restClient, String subscriptionId,
+            AzureTokenCredentials credentials) {
+        return Azure.authenticate(restClient, credentials.domain()).withSubscription(subscriptionId);
```
Comment (chenrujun) [inline sdk/spring/azure-spring-cloud-context/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureContextAutoConfiguration.java:81]: This will throw NPE in original code?

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Comment by saragluna** [comment]
Closing this PR now since the issue has been addressed in https://github.com/Azure/azure-sdk-for-java/pull/17119.
